colorizer_data/converter.py

```diff
@@ -190,40 +214,55 @@ def _write_features(
 def _should_regenerate_frames(
     writer: ColorizerDatasetWriter, data: DataFrame, config: ConverterConfig
 ) -> bool:
-    # get object count and regenerate frames if it has changed
-    num_objects = data[config["object_id_column"]].nunique(False)
 
     if "frames" not in writer.manifest:
         logging.info("No frames found in dataset manifest. Regenerating all frames.")
         return True
     else:
         # Check that all frames exist. If any are missing, frames should be regenerated.
         for frame in writer.manifest["frames"]:
-            if not os.path.exists(writer.outpath / frame):
+            if not os.path.exists(writer.outpath + "/" + frame):
                 logging.info(f"Frame {frame} is missing. Regenerating all frames.")
                 return True
-
+    # get object count and regenerate frames if it has changed
+    num_objects = data[config["object_id_column"]].nunique()
     if writer.manifest["times"] is not None:
         # parse existing times to get object count and compare to new data
         # TODO: refactor this into a utility method for reading json/parquet data
-        times_path = writer.outpath / writer.manifest["times"]
+        times_path = writer.outpath + "/" + writer.manifest["times"]
         _times_filename, times_extension = os.path.splitext(times_path)
-        with open(times_path, "r") as f:
+
+        try:
             times_objects = 0
             if times_extension == ".json":
-                times_json_content = json.load(f)
-                times_objects = len(times_json_content["data"])
+                with open(times_path, "r") as f:
```

Comment:
if it comes up a lot, the check for times_extension seems like it could be abstracted into a little function like get_is_json_file(times_path) or get_file_type